ath79: add support for Belkin F9K1115V2

xdarklight opened the pull request:
This adds support for the Belkin F9K1115V2
It was a supported device in the ar71xx target but has not been ported to ath79 so far.

Support is based on F9K1115 v2 from PR #3380
Many thanks to all who helped to get this in good shape and especially @dmascord for working on that!

There's different ways to write F9K1115V2 (with blank, lower-/upper-case "v" in v2, ...)
I decided to go with what's on the label of the device: F9K1115V2
This can be seen in one of the FCC photos too: https://fccid.io/K7SF9K1115V2/Label/Label-Location-2031947

xdarklight committed:
ath79: make code shared between F9J1108 v2 and F9K1115 v2 re-usable

Belkin F9J1108 v2 and F9K1115 v2 are (seemingly) identical hardware
with different model numbers. Extract all non-device specific code to a
common .dtsi so it can be re-used when adding support for the
F9K1115 v2.

Similar to the .dtsi most of the image buildling recipe code can be
re-used. Move everything except the device model, edimax header magic
and edimax header model into a shared build recipe.

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>

xdarklight committed:
ath79: add support for the Belkin F9K1115 v2 (AC1750 DB Wi-Fi)

This device has (almost?) identical hardware to the F9J1108 v2 but uses
a different firmware magic and model number.

Specifications:

SoC: QCA9558
CPU: 720 MHz
Flash: 16 MiB NOR
RAM: 128 MiB
WiFi 2.4 GHz: QCA9558-AT4A 3x3 MIMO 802.11b/g/n
WiFi 5 GHz: QCA9880-2R4E 3x3 MIMO 802.11a/n/ac
Ethernet: 4x LAN and 1x WAN (all 1Gbit/s ports)
USB: 1 x USB 2.0 (lower), 1 x USB 3.0 (upper)

MAC addresses based on OEM firmware:

Interface   Address   Location
---------   -------   --------
lan         *:5A      sometimes in 0x6
wan         *:5B      0x0
2.4Ghz      *:5A      0x1002
5Ghz        As per mini PCIe EEPROM

Flashing instructions:

The factory.bin can be flashed via the Belkin web UI or via the uboot
HTTP upgrade page (which is by default listening on 192.168.2.1). Once
the factory.bin has been written, sysupgrade.bin will work as usual.

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>

adschm commented:
> qca9558_belkin_f9xyyy-v2.dtsi

This one is missing one "y". Apart from that, I don't really think it's too nice.

Maybe just belkin_f9.dtsi or belkin_f9x-v2.dtsi? Or is there a common board or something that may be more appropriate?

adschm commented on target/linux/ath79/image/generic.mk:
```diff
@@ -402,6 +402,27 @@ define Device/belkin_f9j1108-v2
 endef
 TARGET_DEVICES += belkin_f9j1108-v2
 
+define Device/belkin_f9k1115v2
```
Please separate v2 with a hyphen here, like for the other device.

xdarklight replied:
just to double-check: you still prefer to have the dash (to make it consistent with other boards) even the device label (as state in the PR description) does not use a dash/space?

adschm replied:
I'm under the impression that the v2 represents a version/variant. In that case, it makes sense to separate it as we then could have a separate version field for metadata later without having to touch the name again. If that's not the case, and the v2 stands for something entirely different, one might reconsider.

xdarklight replied:
according to their download page this really is "version 2.0" hardware: https://www.belkin.com/us/support-article?articleNum=42287
it seems like v1 never made to the public.

I'll update the naming to be -v2 as suggested

adschm commented on target/linux/ath79/image/generic.mk:
```diff
@@ -402,6 +402,27 @@ define Device/belkin_f9j1108-v2
+  IMAGES += factory.bin
+  IMAGE/factory.bin := append-kernel | pad-to $$$$(BLOCKSIZE) | \
+	append-rootfs | pad-rootfs | check-size | \
+	edimax-headers eDiMaX F9K1115V2 | pad-to $$$$(BLOCKSIZE)
```
That's a lot of duplicate definition, please create a shared one (there should be enough examples in the file).

xdarklight replied:
OK, I will consolidate this.
do you prefer `Device/belkin_f9x_v2_common` or `Device/belkin_edimax_headers` (or something else)?

adschm replied:
The former. You might even drop the common, just using Device/belkin_f9x_v2 or similar.

Experience tells that being too general doesn't help here (unless you have other devices to join as well, I haven't checked).

adschm commented:
TBH, I don't think we really need the two commits here. But I only have a 70/30 tendency ...

xdarklight commented:
> > qca9558_belkin_f9xyyy-v2.dtsi
> 
> This one is missing one "y". Apart from that, I don't really think it's too nice.

oh, good point

> Maybe just belkin_f9.dtsi or belkin_f9x-v2.dtsi? Or is there a common board or something that may be more appropriate?

I'll go with `belkin_f9x-v2.dtsi` as it shows better that there's some placeholder.
And also it has v2 in it to clarify that it's not any f9... board

xdarklight commented:
v2 of this pull request. changes since v1:
* changed naming to F9K1115 v2 (image building recipe, .dtsi filename, etc.)
* use common code for the image building recipe between F9K1115 v2 and F9J1108 v2

adschm commented:
Thanks, will be merged later today with minor adjustments. Do we really need that `(AC1750 DB Wi-Fi)` in the name, particularly since it's the same for both devices, or could we outsource that to the Wiki page and remove in the code?

xdarklight commented:
> Do we really need that (AC1750 DB Wi-Fi) in the name

I only added it for consistency with other boards. it's the name which is printed on the box
no strong feelings either way from me, so if you decide to drop or keep it: either way is fine with me.
